orientdb/src/test/java/org/hibernte/datastore/ogm/orientdb/test/dialectinvocations/GridDialectOperationInvocationsTest.java

```diff
@@ -0,0 +1,232 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernte.datastore.ogm.orientdb.test.dialectinvocations;
+
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.ogm.cfg.OgmProperties;
+import org.hibernate.ogm.dialect.impl.GridDialects;
+import org.hibernate.ogm.dialect.spi.GridDialect;
+import org.hibernate.ogm.utils.InvokedOperationsLoggingDialect;
+import org.hibernate.ogm.utils.OgmTestCase;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.List;
+import java.util.Map;
+
+import static org.fest.assertions.Assertions.assertThat;
+import static org.junit.Assert.assertNotNull;
+
+/**
+ * Asserts the number and order of grid dialect operations.
+ *
+ * @author cristhiank on 3/6/16 (calovi86 at gmail.com).
+ */
+public class GridDialectOperationInvocationsTest extends OgmTestCase {
```

Comment:
This test is more about the order in which the dialect operations are called it does not depends by the dialect you are using (I think)

A better way to test the dialect is to run the OrientDBTest. That's probably the simplest scenario where we save an entity on the db and the ve load it back.
You will probably need to hack the OrientDBTestHelper class to make it work.

When that pass you can try the to test the BuiltInTypeTest. In these case we want to make sure that the basic java types are recognized by the dialect.